### PR TITLE
feat: integrated address support for Ledger

### PR DESCRIPTION
**Pull request by fluffypony**
## Description
This PR implements integrated address support for the minotari_ledger_wallet application, allowing the Ledger hardware wallet to handle addresses with embedded payment IDs. The implementation supports variable-sized addresses from 67 bytes (standard dual addresses) up to 323 bytes (with maximum 256-byte payment IDs).

### Key Changes:
- Updated address size constants and validation to support variable-length addresses
- Modified address parsing functions to accept `&[u8]` instead of fixed-size arrays
- Added payment ID extraction utility functions
- Enhanced transaction confirmation UI to display payment ID information
- Updated protocol data structures to include address size prefixes
- Added comprehensive test coverage and documentation

## Motivation and Context
Integrated addresses enable users to embed payment identifiers directly within Tari addresses, improving payment tracking and reconciliation. This feature is particularly important for exchanges, merchants, and other services that need to associate payments with specific orders or accounts. The Ledger wallet previously only supported fixed 67-byte dual addresses, limiting its utility for these use cases.

## How Has This Been Tested?
- All existing unit tests pass without modification
- Added comprehensive new test suite covering:
  - Standard dual address compatibility (67 bytes)
  - Integrated addresses with various payment ID sizes
  - Boundary condition testing (min/max sizes)
  - Invalid address size rejection
  - Payment ID extraction and validation
- Compilation verified with `cargo check` and `cargo clippy`
- Both common and comms modules tested successfully

## What process can a PR reviewer use to test or verify this change?
1. **Compilation Check**: Run `cargo check` in `applications/minotari_ledger_wallet/` to verify no compilation errors
2. **Unit Tests**: Run `cargo test` in both `common/` and `comms/` subdirectories to verify all tests pass
3. **Function Verification**: Test the new utility functions:
   - `get_payment_id_bytes_from_tari_dual_address()` with different address sizes
   - `address_has_payment_id()` for detection logic
   - `tari_dual_address_display()` with variable-length inputs
4. **Backward Compatibility**: Verify that existing 67-byte dual addresses continue to work correctly
5. **Edge Cases**: Test with addresses at minimum (67) and maximum (323) byte boundaries

## Breaking Changes
- [x] None
- [ ] Requires data directory on base node to be deleted
- [ ] Requires hard fork
- [ ] Other - Please specify

## Summary by CodeRabbit

- **New Features**
  - Added support for integrated addresses with embedded payment IDs, allowing addresses of variable length.
  - Payment ID information is now displayed during transaction confirmation and review.

- **Bug Fixes**
  - Improved validation and error handling for address lengths and input data during transaction processing.

- **Documentation**
  - Updated README with details on integrated address support, including address size ranges and user confirmation steps.

- **Refactor**
  - Enhanced address handling to support variable-length addresses and streamlined related utility functions.

- **Tests**
  - Added comprehensive tests for address validation, payment ID extraction, and edge cases.

### CODE REVIEW

File: applications/minotari_ledger_wallet/common/src/utils.rs [hunk 1]
```diff
@@ -47,15 +55,35 @@ pub fn tari_dual_address_display(address_bytes: &[u8; TARI_DUAL_ADDRESS_SIZE]) -
...
+    validate_checksum(address_bytes)?;
     let mut public_spend_key_bytes = [0u8; 32];
     public_spend_key_bytes.copy_from_slice(&address_bytes[34..66]);
     Ok(public_spend_key_bytes)
 }
 
+/// Extract payment ID bytes from integrated address, if present
+pub fn get_payment_id_bytes_from_tari_dual_address(address_bytes: &[u8]) -> Result<Vec<u8>, String> {
+    validate_checksum(address_bytes)?;
+    if address_bytes.len() <= TARI_DUAL_ADDRESS_MIN_SIZE {
```
Comment (SWvheerden) [inline applications/minotari_ledger_wallet/common/src/utils.rs:71]:
```suggestion
    if address_bytes.len() = TARI_DUAL_ADDRESS_MIN_SIZE {
```

File: applications/minotari_ledger_wallet/wallet/src/handlers/get_one_sided_metadata_signature.rs [hunk 2]
```diff
@@ -76,10 +78,24 @@ pub fn handler_get_one_sided_metadata_signature(comm: &mut Comm) -> Result<(), A
 
     let commitment_mask: RistrettoSecretKey = get_key_from_canonical_bytes::<RistrettoSecretKey>(&data[40..72])?.into();
 
-    let mut receiver_address_bytes = [0u8; TARI_DUAL_ADDRESS_SIZE]; // 67 bytes
-    receiver_address_bytes.clone_from_slice(&data[72..139]);
+    // Parse variable-length address
+    if data.len() < 74 {
+        return Err(AppSW::WrongApduLength);
+    }
```
Comment (SWvheerden) [inline applications/minotari_ledger_wallet/wallet/src/handlers/get_one_sided_metadata_signature.rs:90]: We should move this check earlier. 
and the which check if data <  171

File: applications/minotari_ledger_wallet/comms/src/accessor_methods.rs [hunk 3]
```diff
@@ -589,13 +589,30 @@ pub fn ledger_get_one_sided_metadata_signature(
...
+    // Check for integrated address support
+    if receiver_address
+        .features()
+        .contains(tari_common_types::tari_address::TariAddressFeatures::PAYMENT_ID)
+    {
+        debug!(
+            target: LOG_TARGET,
+            "Processing integrated address with embedded payment ID"
+        );
+    }
```
Comment (SWvheerden) [inline applications/minotari_ledger_wallet/comms/src/accessor_methods.rs:601]:
```suggestion

```
This seems like debug code?

File: applications/minotari_ledger_wallet/wallet/src/handlers/get_one_sided_metadata_signature.rs [hunk 4]
```diff
@@ -76,10 +78,24 @@ pub fn handler_get_one_sided_metadata_signature(comm: &mut Comm) -> Result<(), A
...
+    let address_size_bytes = &data[72..74]; 
+    let address_size = u16::from_le_bytes([address_size_bytes[0], address_size_bytes[1]]) as usize;
+
+    if address_size < TARI_DUAL_ADDRESS_MIN_SIZE || address_size > TARI_DUAL_ADDRESS_MAX_SIZE {
+        return Err(AppSW::MetadataSignatureFail);
```
Comment (SWvheerden) [inline applications/minotari_ledger_wallet/wallet/src/handlers/get_one_sided_metadata_signature.rs]: Add a correct error here

Comment (SWvheerden) [reply applications/minotari_ledger_wallet/wallet/src/handlers/get_one_sided_metadata_signature.rs]: or reuse : `WrongApduLength`

**Review by SWvheerden — COMMENTED** [review]
Looking good, it needs a few updates and green CI thou

**Review by SWvheerden — APPROVED** [review]
